src/sentry/middleware/devtoolbar.py

```diff
@@ -0,0 +1,70 @@
+import logging
+
+from django.http import HttpRequest, HttpResponse
+
+from sentry import analytics
+from sentry.utils.http import origin_from_request
+from sentry.utils.http import query_string as get_query_string
+from sentry.utils.urls import parse_id_or_slug_param
+
+logger = logging.getLogger(__name__)
+
+
+class DevToolbarAnalyticsMiddleware:
+    def __init__(self, get_response):
+        self.get_response = get_response
+
+    def __call__(self, request):
+        try:
+            response = self.get_response(request)
+        except Exception:
+            response = HttpResponse(status=500)
+            raise  # delegate to our API exception handlers
+        finally:
+            _record_api_request(request, response)
+
+        return response
+
+
+def _record_api_request(request: HttpRequest, response: HttpResponse) -> None:
+    org_id, org_slug, project_id, project_slug = None, None, None, None
+    try:
```

Comment:
i would put this try / except in the caller so this is cleaner to read.

Reply:
Ok, downside is we won't have access to the organization_id, etc kwargs

Reply:
? why would moving the try/catch affect that?

Reply:
i just mean to try/catch the funtion instead of putting it in the logic

Reply:
The function extracts the org_id from the request, which I then log on error

Reply:
remember that sentry is an error monitoring product and we are using it! this information should be included in sentry errors. you do not need to log anything. that is the idea :D

Reply:
Oh haha very true! Thanks

Reply:
I think we'll need sentry_sdk.capture_exc

Reply:
logger.exception will capture sentry errors automatically.

Reply:
but you don't need to include anything in the logs. and if this does ever happen you want to rollback immediately, so you don't need to add analytics information into the logs